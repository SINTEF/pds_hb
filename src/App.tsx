import React from 'react'
import { BrowserRouter, Switch, Route } from 'react-router-dom'
import { Provider, Options } from 'use-http'

import './App.css'
import { Frontpage } from './pages/frontpage'
import { Header } from './components/header'
import { Login } from './pages/login'
import { Read } from './pages/read'
import { NotFound } from './pages/not-found'
import MAIN_ROUTES from './routes/routes.constants'
import { AuthRoute } from './utils/AuthRoute'
import { UserProvider } from './utils/context/userContext'
import useLocalStorage from './utils/hooks/useLocalStorage'

function App(): JSX.Element {
  const { storedValue: token } = useLocalStorage<string>('token', '')

  const options: Partial<Options> = {
    interceptors: {
      // every time we make an http request, this will run 1st before the request is made
      // url, path and route are supplied to the interceptor
      // request options can be modified and must be returned
      request: ({ options }) => {
        options.headers = {
          ...options.headers,
          Authorization: `Bearer ${token}`,
        }

        return options
      },
    },
  }
  return (
    <Provider
      url={process.env.REACT_APP_API_URL || 'http://localhost:5000'}
      options={options}
    >
      <UserProvider>
        <BrowserRouter>
          <Header />
          <Switch>
            <Route path={MAIN_ROUTES.LOGIN}>
              <Login />
            </Route>
            <AuthRoute path={MAIN_ROUTES.READ}>
              <Read />
            </AuthRoute>
            <AuthRoute exact path={MAIN_ROUTES.HOME}>
              <Frontpage />
            </AuthRoute>
            <AuthRoute path={MAIN_ROUTES.NOT_FOUND}>
              <NotFound />
            </AuthRoute>
          </Switch>
        </BrowserRouter>
      </UserProvider>
    </Provider>
  )
}

export default App
