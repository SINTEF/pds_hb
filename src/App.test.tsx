import React from 'react'
import renderer from 'react-test-renderer'
import App from './App'

test('it renders without crashing', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})
