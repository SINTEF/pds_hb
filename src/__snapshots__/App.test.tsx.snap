// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders without crashing 1`] = `
Array [
  <div
    className="container"
  />,
  <div
    className="main"
  >
    <h1>
      PDS Data Handbook
    </h1>
    <div
      className="inputGroup"
    >
      <div
        className="primary  "
      >
        <div
          className="primaryContent"
        >
          <label
            className=""
            htmlFor="Username"
            style={
              Object {
                "alignSelf": "",
              }
            }
          >
            Username
          </label>
          <input
            id="Username"
            onChange={[Function]}
            type="text"
            value=""
          />
          <i
            className="material-icons icon"
          >
            face
          </i>
        </div>
      </div>
      <div
        className="primary  "
      >
        <div
          className="primaryContent"
        >
          <label
            className=""
            htmlFor="Password"
            style={
              Object {
                "alignSelf": "",
              }
            }
          >
            Password
          </label>
          <input
            id="Password"
            onChange={[Function]}
            type="password"
            value=""
          />
          <i
            className="material-icons icon"
          >
            lock
          </i>
        </div>
      </div>
    </div>
    <div
      className="feedbackGroup"
    />
    <div
      className="buttonGroup"
    >
      <div>
        <div
          className="react-ripples"
          onClick={[Function]}
          style={
            Object {
              "display": "inline-flex",
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <button
            className="button medium primary"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": undefined,
              }
            }
            type="button"
          >
            Log in
            <i
              className="material-icons icon"
              style={
                Object {
                  "fontSize": "",
                }
              }
            >
              chevron_right
            </i>
          </button>
          <s
            style={
              Object {
                "borderRadius": "50%",
                "height": 35,
                "opacity": 0,
                "pointerEvents": "none",
                "position": "absolute",
                "transform": "translate(-50%, -50%)",
                "width": 35,
              }
            }
          />
        </div>
      </div>
      <div
        className="react-ripples"
        onClick={[Function]}
        style={
          Object {
            "display": "inline-flex",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <button
          className="button medium standard"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
          type="button"
        >
          Register new user
          <i
            className="material-icons icon"
            style={
              Object {
                "fontSize": "",
              }
            }
          >
            chevron_right
          </i>
        </button>
        <s
          style={
            Object {
              "borderRadius": "50%",
              "height": 35,
              "opacity": 0,
              "pointerEvents": "none",
              "position": "absolute",
              "transform": "translate(-50%, -50%)",
              "width": 35,
            }
          }
        />
      </div>
    </div>
  </div>,
]
`;
